src/backend/wallpaper.rs

```diff
@@ -0,0 +1,143 @@
+use crate::zvariant::{DeserializeDict, OwnedObjectPath, Type};
+use futures_channel::{
+    mpsc::{Receiver, Sender},
+    oneshot,
+};
+use futures_util::{FutureExt, StreamExt};
+use std::cell::RefCell;
+use zbus::{dbus_interface, names::WellKnownName};
+
+use crate::{backend::IMPL_PATH, desktop::wallpaper::SetOn, WindowIdentifierType};
+
+#[derive(DeserializeDict, Type, Debug, Default)]
+#[zvariant(signature = "dict")]
+pub struct WallpaperOptions {
+    #[zvariant(rename = "show-preview")]
+    pub show_preview: Option<bool>,
+    #[zvariant(rename = "set-on")]
+    pub set_on: Option<SetOn>,
+}
+
+pub trait WallpaperImpl {
+    fn set_wallpaper_uri(
+        &self,
+        handle: OwnedObjectPath,
+        app_id: &str,
+        window_identifier: WindowIdentifierType,
+        uri: url::Url,
+        options: WallpaperOptions,
+    ) -> u32;
```

Comment:
```suggestion
    ) -> BasicResponse;
```
instead, make that type public along with `Response` if backend is enabled then, add two constructors to `BasicResponse` for `cancelled`  / `other`  /  `success` or `default`